core/inmem/buffer_test.go

```diff
@@ -0,0 +1,30 @@
+package inmem
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+)
+
+func TestBuffer(t *testing.T) {
+	var wg sync.WaitGroup
+	file := "table1"
+	const len = 4
+
+	append := func(i int) {
```

Comment:
Suggest to use another name  for func. It is not recommended to use `go` reserved names